Src/DependencyCollector/Shared/ComponentCorrelation.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Component Correlation configuration.
+    /// </summary>
+    public class ComponentCorrelation
```

Comment:
I really do not like another public class in this namespace. Also name is not informative. `CrossComponentCorrelationSettings` may sound better. 

I also do not like two classes with the same name in different namespaces.

I'd rather have those properties directly in the module itself.